core/src/main/java/com/distkv/dst/core/concepts/DstSortedListsImpl.java

```diff
@@ -69,34 +54,12 @@ public void incrScore(String key, String member, int delta) {
     if (!dstKeyValueMap.containsKey(key)) {
       throw new KeyNotFoundException(key);
     }
-    LinkedList<SortedListEntity> list = dstKeyValueMap.get(key);
-    ListIterator<SortedListEntity> iterator = list.listIterator();
-    boolean isFound = false;
-    while (iterator.hasNext()) {
-      SortedListEntity now = iterator.next();
-      if (now.getMember().equals(member)) {
-        isFound = true;
-        now.setScore(now.getScore() + delta);
-        if (iterator.nextIndex() != 1) {
-          iterator.remove();
-          while (iterator.hasPrevious()) {
-            SortedListEntity entity = iterator.previous();
-            if (now.compareTo(entity) <= 0) {
-              iterator.add(now);
-              break;
-            }
-            if (now.compareTo(entity) > 0) {
-              iterator.next();
-              iterator.add(now);
-              break;
-            }
-          }
-          break;
-        }
-      }
-    }
-    if (!isFound) {
+    final SortedList sortedList = dstKeyValueMap.get(key);
+    final int isFound = sortedList.incrScore(member, delta);
+    if (0 == isFound) {
       throw new SortedListMemberNotFoundException(key);
+    } else if (-1 == isFound) {
+      throw new SortedListIncrScoreOutOfRange(key);
```

Comment:
Should this be `Member not exists exception`?

Reply:
You can see the method **com/distkv/dst/core/struct/slist/SortedList.java/incrScore**. There is nothing wrong with this place.